src/drivers/esp8266_drv.c

```diff
@@ -405,55 +417,63 @@ static void rx_data_cb(int chan_id, size_t len, const char* data)
 /**
  * Callback that is invoked when the send_data method completes.
  */
-static void _send_data_cb(int bytes_sent)
+static void _send_data_cb(const bool status, const size_t bytes,
+			  const int chan)
```

Comment:
mostly b/c it represents an index of an item, so partially referenced by that article.  Also, you use size_t elsewhere in the code to reference the channel ID, so also to be consistent with that.